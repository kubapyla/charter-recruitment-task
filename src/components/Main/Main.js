import React, { useEffect, useState } from 'react';
import { Container, Button, Stack, Divider } from '@chakra-ui/react';
import { generateDataset } from 'data';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import CustomerTransactionsList from 'components/CustomerTransactionsList';
import CustomersList from 'components/CustomersList';

import customersFixture from 'data/fixtures/customers.json';
import transactionsFixture from 'data/fixtures/transactions.json';

const Main = () => {
  const [dataset, setDataset] = useState({});

  useEffect(() => {
    setDataset({
      customers: customersFixture,
      transactions: transactionsFixture,
    });
  }, []);

  const { transactions, customers } = dataset;

  return (
    <Router>
      <Container maxW={1000} marginBottom={20}>
        <Stack direction={'row'} paddingTop={5} paddingBottom={5}>
          <Button colorScheme="blue" as={Link} to="/">
            Home
          </Button>
        </Stack>
        <Divider />
        <Switch>
          <Route path="/customer/:id">
            <CustomerTransactionsList
              transactions={transactions}
              customers={customers}
            />
          </Route>
          <Route path="/">
            <CustomersList
              customers={customers}
              generateNewDataset={() => setDataset(generateDataset())}
            />
          </Route>
        </Switch>
      </Container>
    </Router>
  );
};

export default Main;
