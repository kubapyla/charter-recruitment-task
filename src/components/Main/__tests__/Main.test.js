import React from 'react';
import { shallow } from 'enzyme';
import Main from '../Main';

describe('Main', () => {
  let wrapper;

  it('should match snapshot', () => {
    wrapper = shallow(<Main />);

    expect(wrapper).toMatchSnapshot();
  });
});
