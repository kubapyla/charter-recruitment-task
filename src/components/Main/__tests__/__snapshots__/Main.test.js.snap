// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
<BrowserRouter>
  <Container
    marginBottom={20}
    maxW={1000}
  >
    <Stack
      direction="row"
      paddingBottom={5}
      paddingTop={5}
    >
      <Button
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        colorScheme="blue"
        to="/"
      >
        Home
      </Button>
    </Stack>
    <Divider />
    <Switch>
      <Route
        path="/customer/:id"
      >
        <CustomerTransactionsList
          customers={Array []}
          transactions={Array []}
        />
      </Route>
      <Route
        path="/"
      >
        <CustomersList
          customers={Array []}
          generateNewDataset={[Function]}
        />
      </Route>
    </Switch>
  </Container>
</BrowserRouter>
`;
